cleverhans/picklable_model.py

```diff
@@ -496,5 +506,36 @@ def set_input_shape(self, shape):
     def get_params(self):
         return []
 
-    def fprop(self, x):
+    def fprop(self, x, **kwargs):
         return tf.map_fn(lambda ex: tf.image.per_image_standardization(ex), x)
+
+
+class Dropout(Layer):
+    """Dropout layer.
+
+    By default, is a no-op. Activate it during training using the kwargs
+    of MLP.fprop.
+    """
+
+    def set_input_shape(self, shape):
+        self.input_shape = shape
+        self.output_shape = shape
+
+    def get_params(self):
+        return []
+
+    def fprop(self, x, dropout_dict=None, **kwargs):
+        """
+        Forward propagation as either no-op or dropping random units.
+        :param x: The input to the layer
+        :param dropout_dict: dict mapping layer names to dropout inclusion
```

Comment:
I am curious what is the reason for using a dictionary common to the model rather than defining the dropout inclusion probabilities at each dropout layer individually?

Reply:
I think i will rewrite it to define the inclusion probability for each layer in that layer's constructor, and then use fprop's kwargs only to say whether to drop or not drop. Does that sound good to you?

Reply:
Yes, that sounds easier to keep track of. Thanks!

Reply:
Done